src/dos/cdrom_ioctl_linux.cpp

```diff
@@ -92,17 +252,35 @@ bool CDROM_Interface_Ioctl::ReadSectors(PhysPt buffer,
 
 bool CDROM_Interface_Ioctl::SetDevice(const char *path, const int cd_number)
 {
-	bool success = CDROM_Interface_SDL::SetDevice(path, cd_number);
+	assert(path != nullptr);
+
+	if (cdrom_fd != -1) {
+		close(cdrom_fd);
+		cdrom_fd = -1;
+	}
+
+	int num = SDL_CDNumDrives();
+	if ((cd_number >= 0) && (cd_number < num)) {
+		const char* cd_name = SDL_CDName(cd_number);
+		if (cd_name) {
+			cdrom_fd = open(cd_name, O_RDONLY | O_NONBLOCK);
+			if (cdrom_fd != -1) {
+				return true;
+			}
+		}
+	}
 
-	if (success) {
-		const char *tmp = SDL_CDName(cd_number);
-		if (tmp)
-			safe_strncpy(device_name, tmp, 512);
-		else
-			success = false;
+	for (int i = 0; i < num; i++) {
+		const char* cd_name = SDL_CDName(i);
+		if (cd_name && strcmp(cd_name, path) == 0) {
+			cdrom_fd = open(cd_name, O_RDONLY | O_NONBLOCK);
+			if (cdrom_fd != -1) {
+				return true;
+			}
+		}
 	}
 
-	return success;
+	return false;
```

Comment:
Ok, so the vertical spacing/grouping is nicely done in this function 🎉  I'd expect the same in all the others.